### PR TITLE
Fix PreferredProperties translation when partitioning on empty set

### LINKED ISSUE #5439: NoSuchElementException in AddExchanges
When running tpcds queries 5, 77, 80, I get the following error:
```
java.util.NoSuchElementException: No value present
	at java.util.Optional.get(Optional.java:135)
	at com.facebook.presto.sql.planner.optimizations.AddExchanges$Rewriter.selectUnionPartitioning(AddExchanges.java:1078)
	at com.facebook.presto.sql.planner.optimizations.AddExchanges$Rewriter.visitUnion(AddExchanges.java:1101)
	at com.facebook.presto.sql.planner.optimizations.AddExchanges$Rewriter.visitUnion(AddExchanges.java:190)
	at com.facebook.presto.sql.planner.plan.UnionNode.accept(UnionNode.java:42)
	at com.facebook.presto.sql.planner.optimizations.AddExchanges$Rewriter.planChild(AddExchanges.java:1209)
	at com.facebook.presto.sql.planner.optimizations.AddExchanges$Rewriter.visitPlan(AddExchanges.java:217)
	at com.facebook.presto.sql.planner.optimizations.AddExchanges$Rewriter.visitPlan(AddExchanges.java:190)
	at com.facebook.presto.sql.planner.plan.PlanVisitor.visitGroupId(PlanVisitor.java:155)
	at com.facebook.presto.sql.planner.plan.GroupIdNode.accept(GroupIdNode.java:125)
	at com.facebook.presto.sql.planner.optimizations.AddExchanges$Rewriter.planChild(AddExchanges.java:1209)
	at com.facebook.presto.sql.planner.optimizations.AddExchanges$Rewriter.visitAggregation(AddExchanges.java:284)
	at com.facebook.presto.sql.planner.optimizations.AddExchanges$Rewriter.visitAggregation(AddExchanges.java:190)
	at com.facebook.presto.sql.planner.plan.AggregationNode.accept(AggregationNode.java:189)
	at com.facebook.presto.sql.planner.optimizations.AddExchanges$Rewriter.planChild(AddExchanges.java:1209)
	at com.facebook.presto.sql.planner.optimizations.AddExchanges$Rewriter.visitProject(AddExchanges.java:233)
	at com.facebook.presto.sql.planner.optimizations.AddExchanges$Rewriter.visitProject(AddExchanges.java:190)
	at com.facebook.presto.sql.planner.plan.ProjectNode.accept(ProjectNode.java:81)
	at com.facebook.presto.sql.planner.optimizations.AddExchanges$Rewriter.planChild(AddExchanges.java:1209)
	at com.facebook.presto.sql.planner.optimizations.AddExchanges$Rewriter.visitTopN(AddExchanges.java:522)
	at com.facebook.presto.sql.planner.optimizations.AddExchanges$Rewriter.visitTopN(AddExchanges.java:190)
	at com.facebook.presto.sql.planner.plan.TopNNode.accept(TopNNode.java:112)
	at com.facebook.presto.sql.planner.optimizations.AddExchanges$Rewriter.planChild(AddExchanges.java:1209)
	at com.facebook.presto.sql.planner.optimizations.AddExchanges$Rewriter.visitOutput(AddExchanges.java:239)
	at com.facebook.presto.sql.planner.optimizations.AddExchanges$Rewriter.visitOutput(AddExchanges.java:190)
	at com.facebook.presto.sql.planner.plan.OutputNode.accept(OutputNode.java:81)
	at com.facebook.presto.sql.planner.optimizations.AddExchanges.optimize(AddExchanges.java:152)
	at com.facebook.presto.sql.planner.LogicalPlanner.plan(LogicalPlanner.java:99)
	at com.facebook.presto.execution.SqlQueryExecution.doAnalyzeQuery(SqlQueryExecution.java:280)
	at com.facebook.presto.execution.SqlQueryExecution.analyzeQuery(SqlQueryExecution.java:259)
	at com.facebook.presto.execution.SqlQueryExecution.start(SqlQueryExecution.java:223)
	at com.facebook.presto.execution.QueuedExecution.lambda$start$1(QueuedExecution.java:62)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
```

Sample query 80:
```
WITH ssr AS
(SELECT
    s_store_id AS store_id,
    sum(ss_ext_sales_price) AS sales,
    sum(coalesce(sr_return_amt, 0)) AS returns,
    sum(ss_net_profit - coalesce(sr_net_loss, 0)) AS profit
  FROM store_sales
    LEFT OUTER JOIN store_returns ON
                                    (ss_item_sk = sr_item_sk AND
                                      ss_ticket_number = sr_ticket_number)
    ,
    date_dim, store, item, promotion
  WHERE ss_sold_date_sk = d_date_sk
    AND d_date BETWEEN cast('2000-08-23' AS DATE)
  AND (cast('2000-08-23' AS DATE) + INTERVAL '30' day)
    AND ss_store_sk = s_store_sk
    AND ss_item_sk = i_item_sk
    AND i_current_price > 50
    AND ss_promo_sk = p_promo_sk
    AND p_channel_tv = 'N'
  GROUP BY s_store_id),
    csr AS
  (SELECT
    cp_catalog_page_id AS catalog_page_id,
    sum(cs_ext_sales_price) AS sales,
    sum(coalesce(cr_return_amount, 0)) AS returns,
    sum(cs_net_profit - coalesce(cr_net_loss, 0)) AS profit
  FROM catalog_sales
    LEFT OUTER JOIN catalog_returns ON
                                      (cs_item_sk = cr_item_sk AND
                                        cs_order_number = cr_order_number)
    ,
    date_dim, catalog_page, item, promotion
  WHERE cs_sold_date_sk = d_date_sk
    AND d_date BETWEEN cast('2000-08-23' AS DATE)
  AND (cast('2000-08-23' AS DATE) + INTERVAL '30' day)
    AND cs_catalog_page_sk = cp_catalog_page_sk
    AND cs_item_sk = i_item_sk
    AND i_current_price > 50
    AND cs_promo_sk = p_promo_sk
    AND p_channel_tv = 'N'
  GROUP BY cp_catalog_page_id),
    wsr AS
  (SELECT
    web_site_id,
    sum(ws_ext_sales_price) AS sales,
    sum(coalesce(wr_return_amt, 0)) AS returns,
    sum(ws_net_profit - coalesce(wr_net_loss, 0)) AS profit
  FROM web_sales
    LEFT OUTER JOIN web_returns ON
                                  (ws_item_sk = wr_item_sk AND ws_order_number = wr_order_number)
    ,
    date_dim, web_site, item, promotion
  WHERE ws_sold_date_sk = d_date_sk
    AND d_date BETWEEN cast('2000-08-23' AS DATE)
  AND (cast('2000-08-23' AS DATE) + INTERVAL '30' day)
    AND ws_web_site_sk = web_site_sk
    AND ws_item_sk = i_item_sk
    AND i_current_price > 50
    AND ws_promo_sk = p_promo_sk
    AND p_channel_tv = 'N'
  GROUP BY web_site_id)
SELECT
  channel,
  id,
  sum(sales) AS sales,
  sum(returns) AS returns,
  sum(profit) AS profit
FROM (SELECT
        'store channel' AS channel,
        concat('store', store_id) AS id,
        sales,
        returns,
        profit
      FROM ssr
      UNION ALL
      SELECT
        'catalog channel' AS channel,
        concat('catalog_page', catalog_page_id) AS id,
        sales,
        returns,
        profit
      FROM csr
      UNION ALL
      SELECT
        'web channel' AS channel,
        concat('web_site', web_site_id) AS id,
        sales,
        returns,
        profit
      FROM wsr) x
GROUP BY ROLLUP (channel, id)
ORDER BY channel, id
LIMIT 100;
```

### PR DESCRIPTION
**Pull request by erichwang**
Fixes #5439

### CODE REVIEW

File: presto-tests/src/main/java/com/facebook/presto/tests/AbstractTestQueries.java [hunk 1]
```diff
@@ -1517,6 +1517,32 @@ public void testGroupingCombinationsDistinct()
     }
 
     @Test
+    public void testRollupOverUnion()
+            throws Exception
+    {
+        MaterializedResult actual = computeActual("" +
+                "SELECT\n" +
```
Comment (electrum) [inline presto-tests/src/main/java/com/facebook/presto/tests/AbstractTestQueries.java]: This SELECT is short, so putting the values on the same line makes it easier to read

File: presto-tests/src/main/java/com/facebook/presto/tests/AbstractTestQueries.java [hunk 2]
```diff
@@ -1517,6 +1517,32 @@ public void testGroupingCombinationsDistinct()
     }
 
     @Test
+    public void testRollupOverUnion()
+            throws Exception
+    {
+        MaterializedResult actual = computeActual("" +
```
Comment (electrum) [inline presto-tests/src/main/java/com/facebook/presto/tests/AbstractTestQueries.java]: I think you can write this as a query and avoid manually creating the result set:
```java
assertQuery(
...,
"VALUES ('P', 21470000), (NULL, 21470000), ...")

**Comment by electrum** [comment]
Minor comments, otherwise looks good

**Comment by electrum** [comment]
If this bug was not introduced in the current release, please add release notes

**Comment by erichwang** [comment]
no release notes necessary. it was introduced in the current release

**Comment by dain** [comment]
You have a test failure
```

Failed tests: 
  TestLocalQueries>AbstractTestQueries.testRollupOverUnion:1542 not equal
Actual 4 rows:
    [O, 439774330]
    [F, 438500670]
    [P, 21470000]
    [null, 899745000]
Expected 5 rows:
    [P, 21470000]
    [null, 21470000]
    [O, 439774330]
    [F, 438500670]
    [null, 878275000]
```

**Pull request merged**